src/AutoRest.CSharp/Output/Models/Types/ObjectType.cs

```diff
@@ -47,19 +54,13 @@ public ObjectType(ObjectSchema objectSchema, BuildContext context): base(context
 
             DefaultAccessibility = objectSchema.Extensions?.Accessibility ?? (hasUsage ? "public" : "internal");
             Description = BuilderHelpers.CreateDescription(objectSchema);
-            DefaultName = objectSchema.CSharpName();
-            if (objectSchema.Extensions?.Namespace is string namespaceExtension)
-            {
-                DefaultNamespace = namespaceExtension;
-            }
-            else if (context.Configuration.ModelNamespace)
+            DefaultName = objectSchema.NameOverride is null ? objectSchema.CSharpName() : objectSchema.NameOverride;
+            if (isResourceModel)
```

Comment:
Use inheritance to override this?

Reply:
the object hierarchy would need to fork at MgmtObjectType here which I felt wasn't necessary given this is a fairly simple if/else let me know if you think that its worth making a whole new class for this

Reply:
Considering we are copying entire types in other places adding an extra type with a simple override should be too complex.